pkg/operator/status.go

```diff
@@ -220,13 +203,95 @@ func (optr *Operator) getOrCreateClusterOperator() (*osconfigv1.ClusterOperator,
 			Namespace: "openshift-config",
 		},
 	}
+}
+
+// defaultStatusConditions returns the default set of stauts conditions for the
+// ClusterOperator resource used on first creation of the ClusterOperator.
+func (optr *Operator) defaultStatusConditions() []osconfigv1.ClusterOperatorStatusCondition {
+	// All conditions default to False with no message.
+	return []osconfigv1.ClusterOperatorStatusCondition{
+		newClusterOperatorStatusCondition(
+			osconfigv1.OperatorProgressing,
+			osconfigv1.ConditionFalse,
+			"", "",
+		),
+		newClusterOperatorStatusCondition(
+			osconfigv1.OperatorDegraded,
+			osconfigv1.ConditionFalse,
+			"", "",
+		),
+		newClusterOperatorStatusCondition(
+			osconfigv1.OperatorAvailable,
+			osconfigv1.ConditionFalse,
+			"", "",
+		),
+	}
+}
+
+// defaultClusterOperator returns the default ClusterOperator resource with
+// default values for related objects and status conditions.
+func (optr *Operator) defaultClusterOperator() *osconfigv1.ClusterOperator {
+	return &osconfigv1.ClusterOperator{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: clusterOperatorName,
+		},
+		Status: osconfigv1.ClusterOperatorStatus{
+			Conditions:     optr.defaultStatusConditions(),
+			RelatedObjects: optr.relatedObjects(),
+		},
+	}
+}
+
+// updateRelatedObjects updates the ClusterOperator's related objects field if
+// necessary and returns the updated ClusterOperator object.
+func (optr *Operator) updateRelatedObjects(co *osconfigv1.ClusterOperator) (*osconfigv1.ClusterOperator, error) {
+	relatedObjects := optr.relatedObjects()
+
 	if !equality.Semantic.DeepEqual(co.Status.RelatedObjects, relatedObjects) {
 		co.Status.RelatedObjects = relatedObjects
 		return optr.osClient.ConfigV1().ClusterOperators().UpdateStatus(co)
 	}
+
 	return co, nil
 }
 
+// getClusterOperator returns the current ClusterOperator.
+func (optr *Operator) getClusterOperator() (*osconfigv1.ClusterOperator, error) {
+	return optr.osClient.ConfigV1().ClusterOperators().
+		Get(clusterOperatorName, metav1.GetOptions{})
+}
+
+// createClusterOperator creates the ClusterOperator and updates its status.
+func (optr *Operator) createClusterOperator() (*osconfigv1.ClusterOperator, error) {
+	defaultCO := optr.defaultClusterOperator()
```

Comment:
fwiw I don't think this would ever be the case given we instantiate via CVO https://github.com/openshift/machine-api-operator/blob/master/install/0000_30_machine-api-operator_12_clusteroperator.yaml

Reply:
Yeah, that's fine. We still need to handle the case of creating it if it's missing for any reason.

Though, maybe it's worth it to check if any of the default conditions are missing in `getOrCreateClusterOperator()` and set the defaults if they are. That would be similar to what we do with the related objects.

Reply:
Done. There's a `setMissingStatusConditions()` method that's called in `getOrCreateClusterOperator()` now. That should mean we handle the case of a missing ClusterOperator and also fully bring any preexisting one that might be missing things up to date.